core/drand_test.go

```diff
@@ -37,385 +37,417 @@ func setFDLimit() {
+		// make sure all nodes had enough time to run their go routines to start the
+		// beacon handler - related to CI problems
+		time.Sleep(getSleepDuration())
+		dt.MoveToTime(group1.GenesisTime)
+		// move to genesis time - so nodes start to make a round
+		// dt.MoveTime(offsetGenesis)
+		// two = genesis + 1st round (happens at genesis)
+		dt.TestBeaconLength(2, false, dt.Ids(oldN, false)...)
+		// so nodes think they are going forward with round 2
+		dt.MoveTime(1 * time.Second)
+
+		// + offline makes sure t
+		toKeep := oldN - offline
+		toAdd := newN - toKeep
+		dt.SetupNewNodes(toAdd, decouplePrevSig)
+
+		fmt.Println("SETUP RESHARE DONE")
+		// run the resharing
+		var doneReshare = make(chan *key.Group)
+		go func() {
+			group, err := dt.RunReshare(toKeep, toAdd, newThr, timeout, false, false)
+			require.NoError(t, err)
+			doneReshare <- group
+		}()
+		time.Sleep(3 * time.Second)
+		fmt.Printf("\n -- Move to Response phase !! -- \n")
+		dt.MoveTime(timeout)
+		// at this point in time, nodes should have gotten all deals and send back
+		// their responses to all nodes
+		time.Sleep(getSleepDuration())
+		fmt.Printf("\n -- Move to Justif phase !! -- \n")
+		dt.MoveTime(timeout)
+		// at this time, all nodes received the responses of each other nodes but
+		// there is one node missing so they expect justifications
+		time.Sleep(getSleepDuration())
+		fmt.Printf("\n -- Move to Finish phase !! -- \n")
+		dt.MoveTime(timeout)
+		time.Sleep(getSleepDuration())
+		// at this time they received no justification from the missing node so he's
+		// exlucded of the group and the dkg should finish
+		// time.Sleep(10 * time.Second)
+		var resharedGroup *key.Group
+		select {
+		case resharedGroup = <-doneReshare:
+		case <-time.After(1 * time.Second):
+			require.True(t, false)
+		}
+		fmt.Println(" RESHARED GROUP:", resharedGroup)
+		target := resharedGroup.TransitionTime
+		now := dt.Now().Unix()
+		// get rounds from first node in the "old" group - since he's the leader for
+		// the new group, he's alive
+		lastBeacon := dt.TestPublicBeacon(dt.Ids(1, false)[0], false)
+		// move to the transition time period by period - do not skip potential
+		// periods as to emulate the normal time behavior
+		for now < target-1 {
+			dt.MoveTime(beaconPeriod)
+			lastBeacon = dt.TestPublicBeacon(dt.Ids(1, false)[0], false)
+			now = dt.Now().Unix()
+		}
+		// move to the transition time
+		dt.MoveToTime(resharedGroup.TransitionTime)
+		time.Sleep(getSleepDuration())
+		// test that all nodes in the new group have generated a new beacon
+		dt.TestBeaconLength(int(lastBeacon.Round+1), true, dt.Ids(newN, true)...)
 	}
-	// move to the transition time
-	dt.MoveToTime(resharedGroup.TransitionTime)
-	time.Sleep(getSleepDuration())
-	// test that all nodes in the new group have generated a new beacon
-	dt.TestBeaconLength(int(lastBeacon.Round+1), true, dt.Ids(newN, true)...)
 }
 
+// FIXME Disable funlen here, but it is failing
+// nolint: funlen
```

Comment:
indication you'll need to break the function up into multiple

Reply:
Yes, I know! However, I just added that flag to skip the error. The func was not changed by us, so I decided not to fix it now, leaving a FIXME flag to see it later on another PR.